src/Datadog.Trace.DuckTyping/DuckTypeExceptions.cs

```diff
@@ -0,0 +1,274 @@
+using System;
+using System.Diagnostics;
+using System.Reflection;
+#pragma warning disable SA1649 // File name must match first type name
+#pragma warning disable SA1402 // File may only contain a single class
+
+namespace Datadog.Trace.DuckTyping
+{
+    /// <summary>
+    /// DuckType Exception
+    /// </summary>
+    public class DuckTypeException : Exception
+    {
+        internal DuckTypeException(string message)
+            : base(message)
+        {
+        }
+    }
+
+    /// <summary>
+    /// DuckType proxy type definition is null
+    /// </summary>
+    public class DuckTypeProxyTypeDefinitionIsNull : DuckTypeException
+    {
+        private DuckTypeProxyTypeDefinitionIsNull()
+            : base($"The proxy type definition is null.")
+        {
+        }
+
+        [DebuggerHidden]
+        internal static void Throw()
+        {
+            throw new DuckTypeProxyTypeDefinitionIsNull();
+        }
+    }
+
+    /// <summary>
+    /// DuckType target object instance is null
+    /// </summary>
+    public class DuckTypeTargetObjectInstanceIsNull : DuckTypeException
+    {
+        private DuckTypeTargetObjectInstanceIsNull()
+            : base($"The target object instance is null.")
+        {
+        }
+
+        [DebuggerHidden]
+        internal static void Throw()
+        {
+            throw new DuckTypeTargetObjectInstanceIsNull();
+        }
+    }
+
```

Comment:
The reason to have that many exceptions for something already covered by other standard exceptions is because I expect to catch the general DuckTypeException in the CallTargetInvoker, when this type of exceptions is thrown, then I will disable the integration, because means that something is wrong with the implementation.

For other exceptions the behavior is to write the log but without disabling the integration completely.

Reply:
That's a good reason